ui/raidboss/data/05-shb/alliance/the_tower_at_paradigms_breach.ts

```diff
@@ -468,6 +476,140 @@ const triggerSet: TriggerSet<Data> = {
+    {
+      id: 'Paradigm Meng-Zi/Xun-Zi Deploy Armaments Trigger',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: ['5C00', '5C01', '5C03', '5C04'], capture: false }),
+      delaySeconds: 0.25,
+      durationSeconds: 5,
+      suppressSeconds: 1,
+      alertText: (data, _matches, output) => {
+        const active = data.deployArmaments;
+
+        if (!active)
+          return;
+
+        if (
+          active.some((e) => e.vertical && !e.sides) &&
+          active.some((e) => !e.vertical && !e.sides)
+        ) {
+          // Two middle-line AOEs, so go to the corner
+          return output.corner!();
+        } else if (
+          active.some((e) => e.vertical && e.sides) &&
+          active.some((e) => !e.vertical && e.sides)
+        ) {
+          // Two side-line AOEs, so go to the center
+          return output.center!();
+        } else if (
+          active.some((e) => e.vertical && !e.sides) &&
+          active.some((e) => !e.vertical && e.sides)
+        ) {
+          // vertical middle-line and horizontal side-lines
+          return output.westBoss!();
+        } else if (
+          active.some((e) => e.vertical && e.sides) &&
+          active.some((e) => !e.vertical && !e.sides)
+        ) {
+          // vertical side-lines and horizontal middle-line
+          return output.northBoss!();
+        } else if (active.some((e) => e.vertical && e.sides)) {
+          // vertical side-lines
+          return output.center!();
+        } else if (active.some((e) => e.vertical && !e.sides)) {
+          // vertical middle-line
+          return output.west!();
+        } else if (active.some((e) => !e.vertical && e.sides)) {
+          // horizontal side-lines
+          return output.center!();
+        } else if (active.some((e) => !e.vertical && !e.sides)) {
+          // horizontal middle-line
+          return output.north!();
+        }
+        // other combinations shouldn't be possible
+        return output.oops!();
+      },
+      outputStrings: {
+        center: {
+          en: 'Go to Center',
+        },
+        northBoss: {
+          en: 'Go to North Boss',
+        },
+        north: {
+          en: 'Go North',
+        },
+        westBoss: {
+          en: 'Go to West Boss',
+        },
+        west: {
+          en: 'Go West',
+        },
+        corner: {
+          en: 'Go to Corner',
+        },
+        oops: {
+          en: 'Avoid line AOEs',
+        },
+      },
+    },
+    {
+      id: 'Paradigm Meng-Zi/Xun-Zi Deploy Armaments Cleanup',
+      type: 'Ability',
+      netRegex: NetRegexes.ability({ id: ['5C00', '5C01', '5C03', '5C04'] }),
+      run: (data, matches) => {
+        if (!data.deployArmaments)
+          return;
+
+        // Get time of current cast
+        const now = Date.parse(matches.timestamp);
+
+        // Filter out any attacks that have completed
+        data.deployArmaments = data.deployArmaments.filter((e) => e.finishedTime > now);
```

Comment:
What do you think about moving this to the start of `Paradigm Meng-Zi/Xun-Zi Deploy Armaments Trigger`? 

Mostly my concern is, what if this trigger happens before the finishedTime and this doesn't get cleaned up? I think if you were filtering out armaments based on their ability id that'd be one thing, but doing it on time seems a little shaky.  Additionally (but minorly), if you did this at the start of one of the other two triggers above, you wouldn't need this third trigger.

Reply:
Ah. Yea that makes sense. When processing a new message there's no reason to care about any that are done when this event is going off. There's no advantage to cleaning htem up earlier and this guarantees we clean them up just before they need to. It avoids any weird pitfalls with logging. Ok cool!